src/contexts/NamespaceContext.ts

```diff
@@ -0,0 +1,20 @@
+import React, { Dispatch, SetStateAction } from 'react';
+import { INamespace } from '../interfaces';
+
+export interface INamespaceContext {
+  selectedNamespace: string;
+  setSelectedNamespace: Dispatch<SetStateAction<string>>;
+  namespaces: INamespace[];
+  setNamespaces: Dispatch<SetStateAction<INamespace[]>>;
+}
+
+export const NamespaceContext = React.createContext<INamespaceContext>({
```

Comment:
This is a good idea to use a context here because it will be probably used in just about every child component that is displaying data